### PR TITLE
Add slides on type casting.

### PR DESCRIPTION
**Pull request by hageboeck**

### CODE REVIEW

File: talk/objectorientation/typecasting.tex [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      \end{cppcode*}
+    \end{exampleblock}
+  \end{overprint}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Type casting}
+  \begin{block}{5 types of casts in \cpp}
+    \begin{itemize}
+    \item \mintinline{c++}{const_cast}: Remove constness from a type. If you think you need this, first try to improve the design!
```
Comment (bernhardmgruber) [inline talk/objectorientation/typecasting.tex:50]:
```suggestion
    \item \mintinline{c++}{const_cast}: Remove constness from a type. If you think you need this, first try to improve your design!
```

**Review by bernhardmgruber — COMMENTED** [review]
we should mention somewhere that `dynamic_cast` only works on polymorphic types, e.i. types with at least 1 virtual member function.

File: talk/objectorientation/typecasting.tex (hunk 1 above)
Comment (hageboeck) [reply talk/objectorientation/typecasting.tex:50]: Most of the time, it wasn't "my" design. Rejected! 🙂

Comment (bernhardmgruber) [reply talk/objectorientation/typecasting.tex:50]: fine

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**